mteb/tasks/Retrieval/dan/DanFeverRetrieval.py

```diff
@@ -0,0 +1,221 @@
+
+                if claim_id not in self.relevant_docs[split]:
+                    self.relevant_docs[split][claim_id] = {}
+
+                self.relevant_docs[split][claim_id][evidence_id] = sim
+
+
+class DanFever(AbsTaskRetrieval):
+    ignore_identical_ids = True
+    superseeded_by = "DanFeverRetrieval"
+
+    metadata = TaskMetadata(
+        name="DanFEVER",
+        dataset={
+            "path": "strombergnlp/danfever",
+            "revision": "5d01e3f6a661d48e127ab5d7e3aaa0dc8331438a",
+            "trust_remote_code": True,
+        },
+        description="A Danish dataset intended for misinformation research. It follows the same format as the English FEVER dataset.",
+        reference="https://aclanthology.org/2021.nodalida-main.47/",
+        type="Retrieval",
+        category="p2p",
+        eval_splits=["train"],
+        eval_langs=["dan-Latn"],
+        main_score="ndcg_at_10",
+        date=("2020-01-01", "2021-12-31"),  # best guess
+        form=["spoken"],
+        domains=["Encyclopaedic", "Non-fiction"],
+        license="CC BY-SA 4.0",
+        socioeconomic_status="mixed",
+        annotations_creators="human-annotated",
+        dialect=[],
+        text_creation="found",
+        bibtex_citation="""
+@inproceedings{norregaard-derczynski-2021-danfever,
+    title = "{D}an{FEVER}: claim verification dataset for {D}anish",
+    author = "N{\o}rregaard, Jeppe  and
+      Derczynski, Leon",
+    editor = "Dobnik, Simon  and
+      {\O}vrelid, Lilja",
+    booktitle = "Proceedings of the 23rd Nordic Conference on Computational Linguistics (NoDaLiDa)",
+    month = may # " 31--2 " # jun,
+    year = "2021",
+    address = "Reykjavik, Iceland (Online)",
+    publisher = {Link{\"o}ping University Electronic Press, Sweden},
+    url = "https://aclanthology.org/2021.nodalida-main.47",
+    pages = "422--428",
+    abstract = "We present a dataset, DanFEVER, intended for multilingual misinformation research. The dataset is in Danish and has the same format as the well-known English FEVER dataset. It can be used for testing methods in multilingual settings, as well as for creating models in production for the Danish language.",
+}
+""",
+        n_samples={"train": 8897},
+        avg_character_length={
+            "train": {
+                "average_document_length": 312.1117274167987,
+                "average_query_length": 50.26957476855484,
+                "num_documents": 2524,
+                "num_queries": 6373,
+                "average_relevant_docs_per_query": 0.48721167425074535,
+            }
+        },
+        task_subtypes=["Claim verification"],
+    )
+
+    def load_data(self, **kwargs):
+        """Load dataset from HuggingFace hub"""
+        if self.data_loaded:
+            return
+        self.dataset = datasets.load_dataset(**self.metadata.dataset)  # type: ignore
+        self.dataset_transform()
+        self.data_loaded = True
+
+    def dataset_transform(self) -> None:
```

Comment:
This function does the same preprocessing fo tboth tasks right? If it's the case, can we factor it into one function outside of the classes and call it in each `data_transform` ?

Reply:
Oh no that is a mistake 2 sec will fix that